src/Model/FileStorage.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Kirby\Model;
+
+use Kirby\Data\Data;
+use Kirby\Exception\DuplicateException;
+use Kirby\Filesystem\F;
+use Kirby\Uuid\Uuid;
+use Throwable;
+
+class FileStorage extends Storage
```

Comment:
`Todo`: Reading the content for a file will need to filter out the `num` and `template` fields as the file handler is the only one that needs to store them in the content file.